tck/src/main/java/ee/jakarta/tck/concurrent/spec/ManagedExecutorService/resourcedef/ManagedExecutorDefinitionOnEJBServlet.java

```diff
@@ -0,0 +1,147 @@
+import ee.jakarta.tck.concurrent.common.context.StringContext;
+import ee.jakarta.tck.concurrent.framework.TestServlet;
+import jakarta.annotation.Resource;
+import jakarta.ejb.EJB;
+import jakarta.enterprise.concurrent.ManagedExecutorService;
+import jakarta.inject.Inject;
+import jakarta.servlet.annotation.WebServlet;
+import jakarta.transaction.UserTransaction;
+
+@WebServlet("ManagedExecutorDefinitionOnEJBServlet")
+public class ManagedExecutorDefinitionOnEJBServlet extends TestServlet {
+	private static final long serialVersionUID = 1L;
+	private static final long MAX_WAIT_SECONDS = TimeUnit.MINUTES.toSeconds(2);
+
+	@Inject AppBean appBean;
+
+	@Resource
+	UserTransaction tx;
+
+	@EJB
+	private ManagedExecutorDefinitionInterface bean;
+
+	/**
+	 * ManagedExecutorService creates an incomplete CompletableFuture to which dependent stages
+	 * can be chained. The CompletableFuture can be completed from another thread lacking the
+	 * same context, but the dependent stages all run with the thread context of the thread
+	 * from which they were created, per ManagedExecutorDefinition config.
+	 */
+	public void testIncompleteFutureEJB() throws Throwable {
+		ManagedExecutorService executor = InitialContext.doLookup("java:app/concurrent/EJBExecutorA");
+
+		try {
+			IntContext.set(181);
+			StringContext.set("testIncompleteFutureEJB-1");
+
+			CompletableFuture<String> stage1 = executor.newIncompleteFuture();
+
+			IntContext.set(182);
+
+			CompletableFuture<String> stage2a = stage1.thenApplyAsync(sep -> {
+				int i = IntContext.get();
+				return "IntContext " + (i == 182 ? "propagated" : "incorrect:" + i) + sep;
+			});
+
+			CompletableFuture<String> stage2b = stage1.thenApply(sep -> {
+				String s = StringContext.get();
+				return "StringContext " + ("".equals(s) ? "cleared" : "incorrect:" + s) + sep;
+			});
+
+			IntContext.set(183);
+
+			CompletableFuture<String> stage3 = stage2a.thenCombineAsync(stage2b, (status1, status2) -> {
+				try {
+					ManagedExecutorService mes = InitialContext.doLookup("java:app/concurrent/EJBExecutorA");
+					return status1 + status2 + "Application context " + (mes == null ? "incorrect" : "propagated");
+				} catch (NamingException x) {
+					throw new CompletionException(x);
+				}                
+			});
+
+			stage1.complete(";");
+
+			String result = stage3.join();
+			assertEquals(result, "IntContext propagated;StringContext cleared;Application context propagated", 
+					"Application context and IntContext must be propagated and StringContext must be cleared " +
+					"per ManagedExecutorDefinition and ContextServiceDefinition config.");
+		} finally {
+			IntContext.set(0);
+			StringContext.set(null);
+		}
+	}
+
+	/**
+	 * ManagedExecutorService can create a contextualized copy of an unmanaged CompletableFuture.
+	 */
+	public void testCopyCompletableFutureEJB() throws Throwable {
+		ManagedExecutorService executor = InitialContext.doLookup("java:app/env/concurrent/executorBRef");
```

Comment:
Will this lookup be valid from the Servlet?  The resource ref name will redirect to a java:module resource that is declared within an EJB,
```
@Resource(lookup = "java:module/concurrent/ExecutorB", name = "java:app/env/concurrent/executorBRef")
```
It would not surprise me if this fails to look up from outside the EJB module.

Reply:
This has been working from the Servlet.

Reply:
Updated to delegate the lookup:
`ManagedExecutorService executor = (ManagedExecutorService) managedExecutorDefinitionBean.doLookup("java:module/concurrent/ExecutorB");
`